build-tools-internal/src/main/groovy/elasticsearch.runtime-jdk-provision.gradle

```diff
@@ -35,8 +35,9 @@ configure(allprojects) {
             test.executable = "${BuildParams.runtimeJavaHome}/bin/java" +
                     (OS.current() == OS.WINDOWS ? '.exe' : '')
         } else {
-            test.dependsOn(project.jdks.provisioned_runtime)
-            test.executable = rootProject.jdks.provisioned_runtime.getBinJavaPath()
+            test.javaLauncher = javaToolchains.launcherFor {
```

Comment:
But we would resolve the exact same version as it would resolve the bundled jdk first. The order of
the resolver decorations in settings.gradle matter.
We can also change it to explicitly request oracle as a vendor. This guarantees we always resolve the exact bundled jdk even in the small chance scenario of temporal network issue with resolving the openjdk from oracle moving on to use adoption instead.

Reply:
Ok, that makes sense. Yeah, I had thought about introducing some kind of custom vendor (i.e. "bundled") that's only resolved by our bundled JDK resolver so there's no risk of using something different. I think if we can give the bundled resolver priority and control that then it's fine.